src/main/java/org/jabref/gui/preferences/preview/PreviewTabViewModel.java

```diff
@@ -92,9 +92,10 @@ public PreviewTabViewModel(DialogService dialogService, PreferencesService prefe
         initialPreviewPreferences = preferences.getPreviewPreferences();
 
         sourceTextProperty.addListener((observable, oldValue, newValue) -> {
-            var currentLayout = getCurrentLayout();
-            if (currentLayout instanceof TextBasedPreviewLayout) {
-                ((TextBasedPreviewLayout) currentLayout).setText(sourceTextProperty.getValue().replace("\n", "__NEWLINE__"));
+            try {
+                layoutParse();
+            } catch (StringIndexOutOfBoundsException e) {
+                // catch this error but not give to user [when typing]
```

Comment:
Catching runtime excpetions is a very bad thing. Where exactly is the `StringIndexOutOfBoundsException` thrown? Exceptions should be catched as early as possible.

In this case, it is thrown at `parse()` in LayoutHelper.java line 191.

The real fix is that `org.jabref.logic.layout.LayoutHelper#getLayoutFromText` should return `Optional<Layout>` and not throw any exception.

Reply:
Hello, sir. I am also following this issue. Do you think this exception `StringIndexOutOfBoundsException ` is unnecessary? This `StringIndexOutOfBoundsException ` will be thrown when '\' is entered(in editing stage). 
What if it's in the saving stage?  If user click the `save` button, and there is only a '\', should the exception still be thrown?
If we do not throw any exception, the user may not be aware that there is a wrong '\'. If you insist that throwing no exception is a better choice, I can help you implement it. Don't worry about @yinpeiqi and @zc-BEAR. I have communicated with them. Looking forward to your reply.

Reply:
After my investigation, `Layout` could not be null. Therefore `LayoutHelper#getLayoutFromText` seems not effective. In my opinion, we can delete the code
`throw new StringIndexOutOfBoundsException(
                            "Backslash parsing error near \'" + lastFive.toString().replace("\n", " ") + '\'');` in `LayoutHelper#parseField`.
I also found that `Layout`'s constructor will  warn "Nested field/group entries are not implemented!" in logger if the user input's invalid.  In this case, does the error message need to be presented to the user? @koppor

Reply:
It's not about `Layout` being `null`, its about getting rid of the exception. Using exceptions for control flow is a bad thing (see also Java-by-Comparison). Please refactor `LayoutHelper` not to throw any exception at `parse()` or `getLayoutFromText()`.

Reply:
@EricLee543 I am not sure how to present that to the user. In the long run, we should switch to Apache Velocity as this shifts maintainence effort from JabRef to another project. See https://github.com/koppor/jabref/issues/392 for details.